lib/graphql/schema.rb

```diff
@@ -358,6 +358,14 @@ def middleware
       end
     end
 
+    def middleware_chain
+      @middleware_chain ||= begin
+        steps = middleware.dup
+        steps << GraphQL::Execution::Execute::FieldResolveStep
+        GraphQL::Schema::MiddlewareChain.new(steps: steps)
+      end
```

Comment:
If middleware can be changed _after_ the schema is defined, this variable would have to be cleared after each change.